app/main.py

```diff
@@ -1,5 +1,82 @@
+from __future__ import annotations
+
+
 class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        if coords is None:
+            coords = [0, 0]
+        self.name = name
+        self.weight = weight
+        self.coords = coords
+
+    def go_forward(self, step: int = 1) -> list:
+        x, y, *z = self.coords
+        y += step
+        self.coords = [x, y, *z]
+        return self.coords
```

Comment:
You could simplify the logic by making the in-place list modification, e.g.
```suggestion
        self.coords[1] += step
```
Apply this change to all methods that modify coordinates for all classes.